src/core/mac/mac_frame.cpp

```diff
@@ -998,6 +1000,40 @@ void TxFrame::CopyFrom(const TxFrame &aFromFrame)
 #endif
 }
 
+void TxFrame::ProcessTransmitAesCcm(const ExtAddress &aExtAddress)
+{
+#if OPENTHREAD_RADIO
```

Comment:
:+1:  for adding this method in `TxFrame` (moving it from `Mac`). I think this makes great sense.

One suggestion here: How about add `#ifdef OPENTHREAD_MTD || OPENTHREAD_FTD` to the whole definition/implementation of this method (i.e., not have it even defined under `OPENTHREAD_RADIO`).  This way, if it happens to be used under `OPENTHREAD_RADIO` (by mistake) we get a linker error.

Reply:
Thanks for the quick review. We cannot do that easily because the platform library is build without knowing the upper layer is rcp or not. All ftd, mtd, rcp link to the same platform library. We can introduce another special platform library for RCP, but i tend it to be done in a separate PR. Thoughts?

Reply:
How about keeping the declaration but not having any implementation of this method under `OPENTHREAD_RADIO`? The platform lib should be happy seeing a declartion of this method and we still would get a linker error if used?

Reply:
I'm afraid not. ot-rcp is build together with ot-ncp and ot-cli. It must link all symbols.

Reply:
Ok. sounds good.